lib/system/gc.nim

```diff
@@ -445,6 +445,19 @@ proc gcInvariant*() =
     markForDebug(gch)
 {.pop.}
 
+template setFrameInfo(c: PCell) =
+  when leakDetector:
+    if framePtr != nil and framePtr.prev != nil:
+      var fr = framePtr.prev
+      while not fr.prev.isNil and
+          (c_strcmp(fr.filename, "system.nim") == 0 or c_strcmp(fr.filename, "gc.nim") == 0):
```

Comment:
with -d:excessiveStackTraces or similar 'filename' can be a full path. I also think it's dangerous to throw stack trace information away.